user/userns/userns_linux.go

```diff
@@ -0,0 +1,49 @@
+package userns
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"sync"
+)
+
+var inUserNS = sync.OnceValue(runningInUserNS)
+
+// runningInUserNS detects whether we are currently running in a user namespace.
+//
+// Originally copied from https://github.com/lxc/incus/blob/e45085dd42f826b3c8c3228e9733c0b6f998eafe/shared/util.go#L678-L700.
```

Comment:
nit: we should probably refer to runc, too.

Reply:
Added a mention of runc (both here, and in the package description)